app/src/main/java/com/vci/vectorcamapp/core/data/upload/image/ImageUploadWorker.kt

```diff
@@ -2,85 +2,508 @@ package com.vci.vectorcamapp.core.data.upload.image
+                    successfulUploadsInARow = 0
+                    currentChunkSize =
+                        (currentChunkSize / FAILURE_CHUNK_SIZE_DIVIDER).coerceAtLeast(
+                            MIN_CHUNK_SIZE_BYTES
+                        )
+                    Log.w(
+                        "ImageUploadWorker",
+                        "Upload error: $error. Reducing chunk size to $currentChunkSize bytes."
+                    )
+
+                    if (error != NetworkError.REQUEST_TIMEOUT && error != NetworkError.NO_INTERNET && error != NetworkError.SERVER_ERROR) {
+                        return DomainResult.Error(error)
+                    }
+
+                    recoveryAttempts++
+
+                    if (recoveryAttempts >= MAX_RETRIES) {
+                        Log.e(
+                            "ImageUploadWorker",
+                            "Exceeded max recovery attempts for a single image upload. Failing."
+                        )
+                        return DomainResult.Error(error)
+                    }
+
+                    try {
+                        uploader = client.resumeOrCreateUpload(upload)
+                        continue
+                    } catch (resumeException: Exception) {
+                        Log.e("ImageUploadWorker", "Failed to resume upload.", resumeException)
+                        return DomainResult.Error(NetworkError.UNKNOWN_ERROR)
+                    }
+                }
+            }
+        }
+        Log.d("ImageUploadWorker", "Upload loop finished. Final offset: ${uploader.offset}")
+        return DomainResult.Success(Unit)
+    }
+
+    private fun createTusUpload(
+        file: File,
+        fingerprint: String,
+        contentType: String?,
+        md5: String
+    ) = TusUpload(file).apply {
+        this.fingerprint = fingerprint
+        metadata = mapOf(
+            "filename" to file.name,
+            "contentType" to (contentType ?: "application/octet-stream"),
+            "filemd5" to md5
         )
+    }
+
+    private fun calculateMD5(file: File): String {
+        val md = MessageDigest.getInstance("MD5")
+        FileInputStream(file).use { fis ->
+            val buffer = ByteArray(BYTE_ARRAY_SIZE)
+            var read: Int
+            while (fis.read(buffer).also { read = it } != -1) {
+                md.update(buffer, 0, read)
+            }
+        }
+        return md.digest().joinToString("") { "%02x".format(it) }
+    }
+
+    private suspend fun prepareFile(resolver: ContentResolver, source: Uri): Pair<File, String?> =
+        withContext(Dispatchers.IO) {
+            val mimeType = resolver.getType(source)
+            val extension = when (mimeType) {
+                "image/jpeg" -> "jpg"
+                "image/png" -> "png"
+                else -> "bin"
+            }
+            val destination =
+                File(context.cacheDir, "work_upload_${System.currentTimeMillis()}.$extension")
```

Comment:
You should add a check in place to ensure that if the image has not already been added to the cache, only then it should be added. Is there risk of duplicate copies in the cache?